pydatastructs/miscellaneous_data_structures/_backend/cpp/stack/stack.cpp

```diff
@@ -0,0 +1,48 @@
+#include <Python.h>
+#include "ArrayStack.hpp"
+
+static struct PyModuleDef stack_struct = {
+    PyModuleDef_HEAD_INIT,
+    "_stack",
+    0,
+    -1,
+    NULL,
+};
+
+PyMODINIT_FUNC PyInit__stack(void) {
+    Py_Initialize();
+    PyObject *stack = PyModule_Create(&stack_struct);
+
+    if (PyType_Ready(&ArrayStackType) < 0) {
+        return NULL;
+    }
+    Py_INCREF(&ArrayStackType);
+    PyModule_AddObject(stack, "ArrayStack", reinterpret_cast<PyObject*>(&ArrayStackType));
+
+    if (PyType_Ready(&ArrayType) < 0) {
+        return NULL;
+    }
+    Py_INCREF(&ArrayType);
+    PyModule_AddObject(stack, "Array", reinterpret_cast<PyObject*>(&ArrayType));
+
+    if (PyType_Ready(&OneDimensionalArrayType) < 0) {
+        return NULL;
+    }
+    Py_INCREF(&OneDimensionalArrayType);
+    PyModule_AddObject(stack, "OneDimensionalArray", reinterpret_cast<PyObject*>(&OneDimensionalArrayType));
+
+    if (PyType_Ready(&DynamicArrayType) < 0) {
+        return NULL;
+    }
+    Py_INCREF(&DynamicArrayType);
+    PyModule_AddObject(stack, "DynamicArray", reinterpret_cast<PyObject*>(&DynamicArrayType));
+
+    if (PyType_Ready(&DynamicOneDimensionalArrayType) < 0) {
+        return NULL;
+    }
+    Py_INCREF(&DynamicOneDimensionalArrayType);
+    PyModule_AddObject(stack, "DynamicOneDimensionalArray", reinterpret_cast<PyObject*>(&DynamicOneDimensionalArrayType));
```

Comment:
The segmentation was being thrown because the array types weren't registered in the `stack` extension module. So there are two options,

1. Register them manually (as I did in the above). 
2. Import them at runtime via the APIs in https://docs.python.org/3/c-api/import.html.

For now approach in point 1 seems much easier and efficient because we pre-register everything that's going to be used in the module's implementation. So no need to import them during function calls. The second approach is messier as well (`PyImport_Import` every now and then whenever we want to use something from another C++ extension module).

Reply:
The following code works for me for example with 4b028f158556f70ebc8bb144d0995b1faea70b4f commit.

```python
from pydatastructs.miscellaneous_data_structures._backend.cpp import _stack

array_stack = _stack.ArrayStack([1, 2, 3], int)
```